### PR TITLE
[DOM Reference]Update Attr interface reference and add missing properties

### PR DESCRIPTION
**Pull request by teoli2003**
Part of #9740

- Update `Attr` and the existing children `localName`, `namespaceURI`, `prefix`.
- Create the missing children `name`, `ownerElement`, `specified`, and `value`.

Note to the reviewer: one key change is to use a coherent explanation of namespaces. Namespaces are not existing in HTML5 (unlike XHTML, but like former HTML), except for SVG and MathML embedded in pages. This was not explained (and still is fuzzy in Element, but this will be fixed when I deal with this interface).

So a good test is: do you understand my explanations.

I will update spec_url and mdn_url in bcd too. PR mdn/browser-compat-data#13109

### CODE REVIEW

**Review by dominiccooney — COMMENTED** [review]
Wow. The existing content was truly ancient. This is a great update. Some suggestions inline.

File: files/en-us/web/api/attr/index.md [hunk 1]
```diff
@@ -2,113 +2,44 @@
...
-  - API
-  - DOM
+  - Interface
+  - Reference
 browser-compat: api.Attr
 ---
 {{APIRef("DOM")}}
 
-The **`Attr`** interface represents one of a DOM element's attributes as an object. In most DOM methods, you will directly retrieve the attribute as a string (e.g., {{domxref("Element.getAttribute()")}}), but certain functions (e.g., {{domxref("Element.getAttributeNode()")}}) or means of iterating return `Attr` types.
+The **`Attr`** interface represents one of an element's attributes as an object. In most methods, you will directly retrieve the attribute as a string (e.g., {{domxref("Element.getAttribute()")}}), but certain functions (e.g., {{domxref("Element.getAttributeNode()")}}) or means of iterating return `Attr` types.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/attr/index.md:11]: Fixed

File: files/en-us/web/api/attr/index.md [hunk 2]
```diff
@@ -2,113 +2,44 @@
...
-  - : A {{domxref("DOMString")}} representing the namespace URI of the attribute, or `null` if there is no namespace.
-- {{domxref("Attr.localName", "localName")}} {{readOnlyInline}}
-  - : A {{domxref("DOMString")}} representing the local part of the qualified name of the attribute.
-- {{domxref("Attr.prefix", "prefix")}} {{readOnlyInline}}
-  - : A {{domxref("DOMString")}} representing the namespace prefix of the attribute, or `null` if no prefix is specified.
+  - : A {{jsxref("String")}} representing the URI of the namespace of the attribute, or `null` if there is no namespace.
 - {{domxref("Attr.ownerElement", "ownerElement")}} {{readOnlyInline}}
+  - : The {{domxref("Element")}} the attribute belongs to.
+- {{domxref("Attr.prefix", "prefix")}} {{readOnlyInline}}
+  - : A {{jsxref("String")}} representing the namespace prefix of the attribute, or `null` if no namespace is specified.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/attr/index.md]: Ha! This was it, I missed the point that a namespace without prefix could exist. Fixed.

File: files/en-us/web/api/attr/index.md [hunk 3]
```diff
@@ -2,113 +2,44 @@
...
+The **`Attr`** interface represents one of an element's attributes as an object. In most methods, you will directly retrieve the attribute as a string (e.g., {{domxref("Element.getAttribute()")}}), but certain functions (e.g., {{domxref("Element.getAttributeNode()")}}) or means of iterating return `Attr` types.
+
+The core idea of an object of type `Attr` is the association between a _name_ and a _value_. An attribute may also be part of a _namespace_ and, in this case, it also have a URI identifying the namespace, and a prefix that is an abbreviation for the namespace.
+
+The name is deemed _local_ when it ignores the eventual namespace and deemed _qualified_ when it includes the prefix of the namespace, separated from the local name by a colon (`:`). For example, if we have two attributes called `myAttr`, one without a namespace, one inside the `example` namespace, we have: `myAttr` for the qualified name of the attribute without a namespace and `example:myAttr` for the qualified name of the attribute within the `example` namespace. Both have the same local name, `myAttr`.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/attr/index.md]: I tried to rewrite and put a table. I missed the possibility of a namespace without a prefix.

File: files/en-us/web/api/attr/localname/index.md [hunk 4]
```diff
@@ -2,61 +2,63 @@
...
-## Syntax
+The local name is always in lower case, whatever case at the attribute creation.
 
-```js
-name = attribute.localName
-```
+> **Note:** Only XML supports namespaces. HTML does not. That means that the local name of an attribute of an HTML element is always be equal to its qualified name: Colons are treated as regular characters. In XML, like in SVG or MathML, the colon denotes the end of the prefix and what is before is the namespace; the local name may be different from the qualified name.
 
-### Return value
+Also, only the `xml` (for the `xml:lang` attribute), `xlink` (for the `xlink:href`, `xlink:show`, `xlink:target` and `xlink:title` attributes) and `xpath` namespaces are supported, and only on SVG and MathML elements.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/attr/localname/index.md]: Yes, fixed.

**Comment by teoli2003** [comment]
I think I have addressed all comments. Thanks for the review. Namespaces are definitively a kind of oddity in HTML.

**Comment by teoli2003** [comment]
@dominiccooney (I can't re-request a review from the GitHub UI, hence this comment) Can you check again when you have some time? Thanks.

**Review by Elchi3 — APPROVED** [review]
Nice work! 👍

**Pull request merged**